### ISSUE TITLE
Ensure that Pixeltable is robust against multiple concurrent clients

### ISSUE THREAD

**Issue by dgutman** [pixeltable/pixeltable#355]
Given my level of inpatience, I was trying to work on some downstream analysis in a separate jupyter notebook, while loading /parsing some data in the initial notebook.

import numpy as np
import pixeltable as pxt
import os
from pixeltable.functions.audio import get_metadata
from pixeltable.functions import whisper

audio_table = pxt.get_table('demo.audio_table')

The only way I could get this second notebook to be aware of additional columns I added were to actually restart the kernel, simply rerunning the get_table('demo.audio_table') did not seem to update anything.

**Comment by mkornacker** [comment]
That's correct - currently, the catalog information is cached internally. One more thing to keep in mind: at this point, Pixeltable does not support concurrent update operations (ie, insert/update/delete) to the same table in separate notebooks, also related to the how catalog information is being cached today.

We're planning on changing that sometime soon, though.

As a short-term workaround, you can do the following (but keep in mind that this is an internal interface):
pxt.catalog.Catalog.clear()
pxt.init()

**Comment by dgutman** [comment]
I figured there was some layer of cacheing.. Just thinking about cases
where there is some large data load process and inpatient devs want to
start doing some preliminary analysis / sanity checking.

**Comment by aaron-siegel** [comment]
This is currently planned for 2025-Q1.
